### PR TITLE
Support for ConsumerRecords container type on batch processing

### PR DESCRIPTION
**Pull request by rorueda**
See #903

The changes are:
- Bind arguments of `ConsumerRecords` type;
- Consider `ConsumerRecords` type when looking for the "body argument" to configure the correct key and value deserializers;
- Add a test;

### CODE REVIEW

**Comment by sdelamo** [comment]
@jeremyg484 can you please review this?

File: kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java [hunk 1]
```diff
@@ -544,20 +549,20 @@ private static boolean isLastArgumentOfSuspendedMethod(Argument<?> argument, Exe
...
-        final Argument<?> bodyArgument = findBodyArgument(batch, method);
+        final Optional<Argument<?>> bodyArgument = findBodyArgument(batch, method);
         configureKeyDeserializer(bodyArgument, method, config);
         configureValueDeserializer(bodyArgument, config);
         debugDeserializationConfiguration(method, config);
     }
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
-    private void configureKeyDeserializer(Argument<?> bodyArgument, ExecutableMethod<?, ?> method, DefaultKafkaConsumerConfiguration config) {
+    private void configureKeyDeserializer(Optional<Argument<?>> bodyArgument, ExecutableMethod<?, ?> method, DefaultKafkaConsumerConfiguration config) {
```
Comment (jeremyg484) [inline kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java:559]: We generally try not to use `Optional` as a method parameter (though I know there are some cases in this project where that has been violated).

Can you refactor this to use the previous signature without `Optional`?

File: kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java [hunk 2]
```diff
@@ -567,25 +572,26 @@ private void configureKeyDeserializer(Argument<?> bodyArgument, ExecutableMethod
     }
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
-    private void configureValueDeserializer(Argument<?> bodyArgument, DefaultKafkaConsumerConfiguration config) {
+    private void configureValueDeserializer(Optional<Argument<?>> bodyArgument, DefaultKafkaConsumerConfiguration config) {
```
Comment (jeremyg484) [inline kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java:575]: We generally try not to use `Optional` as a method parameter (though I know there are some cases in this project where that has been violated).

Can you refactor this to use the previous signature without `Optional`?

**Review by jeremyg484 — CHANGES_REQUESTED** [review]
The idea and code itself are great, other than my minor nitpicking about not using `Optional` as a method parameter.

Beyond that, the docs need to be updated to describe this new option.

**Comment by rorueda** [comment]
I rebased it onto master and made the requested change.

**Pull request merged**